src/main/java/net/neoforged/neoforge/common/extensions/IItemStackExtension.java

```diff
@@ -490,4 +494,61 @@ default ItemAttributeModifiers getAttributeModifiers() {
 
         return CommonHooks.computeModifiedAttributes(self(), defaultModifiers);
     }
+
+    /**
+     * Wraps the original {@link ItemStack#save} encoding behavior and logs additional information if an exception is thrown.
+     */
+    default Tag encode(HolderLookup.Provider provider, Tag tag) {
+        try {
+            return self().CODEC.encode(self(), provider.createSerializationContext(NbtOps.INSTANCE), tag).getOrThrow();
```

Comment:
`CODEC` is a `public static` field, `self()` here feels unnecessary, should be `ItemStack.CODEC` instead.
[_Same for line 515_]